src/core/src/sys/energy.rs

```diff
@@ -29,10 +30,10 @@ impl<'a> EnergyEvaluator<'a> {
     /// Compute the energy associated with the pair of particles `i, j` at
     /// distance `r`
     #[inline]
-    pub fn pair(&self, r: f64, i: usize, j: usize) -> f64 {
-        let distance = self.system.bond_distance(i, j);
+    pub fn pair(configuration: &Configuration, potentials: LocalPotentials, r: f64, i: usize, j: usize) -> f64 {
```

Comment:
Because I did not consider the configuration to be part of the public API for `LocalPotentials`, I agree it's strange.